doc: Add basic service documentation (#36)
epiphany-platform/CommonDataLayer: docs/Command-Service.md

# Command Service

### Responsibilities

> The command service is reponsible for consuming a specific input topic and writing that data to its to a variety of different stroage solutions. 

Reviewer: ```suggestion
The command service is responsible for consuming a specific input topic and writing that data to a variety of different storage solutions. 
```

> The command service is responsible for consuming a specific input topic and writing that data to its to a variety of different storage solutions. 

Reviewer 2: ```suggestion
The command service is responsible for consuming a specific input topic and writing that data to a variety of different storage solutions. 
```

### Communication

> Communication from command service is done through through Kafka. An input topic is consumed and the message is routed to the corresponding db based on configuration. 

Reviewer: ```suggestion
Communication from the command service is done through through Kafka. An input topic is consumed and the message is routed to the corresponding database based on configuration. 
```

### Configuration

It is important to ensure that configure command service with the url to a database from eitehr Postgres, Druid or SLED

```
KAFKA_INPUT_GROUP_ID
KAFKA_INPUT_BROKERS
KAFKA_INPUT_TOPIC: 

> POSTGRES_OUTPUT_URL

Reviewer 3: Well, this has changed. We had to split POSTGRES config into separate envs for `DE` deployment. 
https://github.com/epiphany-platform/CommonDataLayer/blob/develop/examples/deploy/docker-compose.yml#L103

Reviewer 4: ```suggestion
POSTGRES_USERNAME
POSTGRES_PASSWORD
POSTGRES_HOST
POSTGRES_PORT
POSTGRES_DBNAME
POSTGRES_SCHEMA
```

# Command Service

### Technical Description

The Command-Service (commonly refered also as `CS`, or `CSPG` - indicating posgres instance), interfaces storage repositories with the CDL ecosystem.

Interacts with:
- Data Router (optional, either)
- Message Queue (optional, either)
- Supported Repository (one of)

Ingest methods:
- Kafka
- RabbitMq
- GRPC (currently either only one instance without kubernetes)

Egest methods (supported repositores):
- Postgresql (tested on 12, should support anything >=9, advised 13)
- VictoriaMetrics
- Druid
- Sleight (CDL's document storage)
- Troika (CDL's binary data repo)
- .. or anything with matching GRPC :)

### Configuration (environment files)

```
KAFKA_INPUT_GROUP_ID
KAFKA_INPUT_BROKERS

> KAFKA_INPUT_TOPIC: 

Reviewer 4: ```suggestion
KAFKA_INPUT_TOPIC
```
